schema.graphql

```diff
@@ -91,6 +91,10 @@ type Mutation {
   addPallet(input: PalletCreateInput!): Pallet!
   updatePallet(id: Int!, input: PalletUpdateInput!): Pallet!
   destroyPallet(id: Int!): Offer! # return the destroyed pallet's offer
+  addLineItem(palletId: Int!): Pallet! # return the pallet the line item was added to
```

Comment:
Thought: returning a `Pallet` feels weird because I would normally expect a `LineItem`. What's a use case for this?

Reply:
The line item is displayed in the context of the pallet as an entry in its list of line items. I figured returning the pallet would allow the mutation caller to specify a query that includes the list of line items for that pallet so upon success can update the client-side pallet's line item list in entirety rather than attempting to book-keep adding the new line item. My guess is that would make things simpler on the client. Another plus would be if two people were editing the same offer at the same time their clients would be less likely to get out of sync.

Reply:
Based on what I'm reading in the [blog post you linked](https://www.apollographql.com/blog/when-to-use-refetch-queries-in-apollo-client/), it sounds like this behavior could make things easier with managing the cache:
> One way we could omit needing to write this update function is if the mutation response returned the entire list of new todos in the mutation response, but that’s not always possible.

Reply:
Those are very good reasons 👍